client/src/games/cafe/MainMenu.tsx

```diff
@@ -6,91 +6,290 @@ The full terms of this copyright and license should always be found in the root
 */
 
 import Phaser from "phaser";
-import { addImage, addSprite, addText, scaleImage } from "../phaser-helpers";
-import { CafeGameParams } from "./MainGame";
+import {
+  addBackground,
+  addImage,
+  addSprite,
+  addText,
+  Anchor,
+  scaleImage,
+} from "../phaser-helpers";
+import { randomInt } from "../../utils";
+import { SPAWN_TIME } from "./simulator";
+import { GameParams } from "games";
 
 export default class MainMenu extends Phaser.Scene {
-  text?: Phaser.GameObjects.Text;
-  timerText?: Phaser.GameObjects.Text;
-  accuracyText?: Phaser.GameObjects.Text;
-  images: Phaser.GameObjects.Image[];
+  bg?: Phaser.GameObjects.Image;
+  screen?: Phaser.GameObjects.Image;
+  screenText?: Phaser.GameObjects.Text;
+  pal?: Phaser.GameObjects.Sprite;
+  trash: Phaser.GameObjects.Sprite[];
+  trashText: Phaser.GameObjects.Text[];
+  scoreText?: Phaser.GameObjects.Text;
 
   constructor() {
     super("MainMenu");
-    this.images = [];
+    this.trash = [];
+    this.trashText = [];
   }
 
   preload() {
+    this.load.setPath("assets");
+    this.load.spritesheet("pal", "pal_spritesheet.png", {
+      frameWidth: 250,
+      frameHeight: 250,
+    });
+    this.load.image(
+      "window",
+      "wordui/inner_panels/blue/premade_panels/blue_long_horizontal.png"
+    );
+    this.load.image(
+      "button",
+      "wordui/buttons/long_buttons/blue_button_complete.png"
+    );
     this.load.setPath("assets/cafe");
-    this.load.image("background", "background.png");
-    this.load.setPath("assets/cafe/sprites");
-    this.load.atlas("bg_kitchen", "bg_kitchen.png", "bg_kitchen.json");
-    this.load.atlas("char_bears", "char_bears.png", "char_bears.json");
-    this.load.atlas("char_speech", "char_speech.png", "char_speech.json");
-    this.load.atlas("food", "food.png", "food.json");
+    for (let i = 1; i <= 2; i++) {
+      this.load.image(`background${i}`, `background${i}.jpg`);
+    }
+    for (let i = 1; i <= 4; i++) {
+      this.load.image(`bin${i}`, `bin${i}.png`);
+    }
+    for (let i = 1; i <= 4; i++) {
+      this.load.image(`zap${i}`, `zap${i}.png`);
+    }
+    for (let i = 1; i <= 10; i++) {
```

Comment:
I see a fair amount of re-used code between MainMenu.tsx and MainGame.tsx. If it's possible, might be worth implementing some way to share the code, although it looks like that might be easier said than done